### PR TITLE
Introduce `random` JsonPath function

### PR DESCRIPTION
**Pull request by uarlouski**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -73,6 +73,9 @@ The function output is dictated by the function itself.
 | `last()`    | Provides the last item of an array                                                   | Depends on the array |
 | `index(X)`  | Provides the item of an array of index: X, if the X is negative, take from backwards | Depends on the array |
 | `distinct()`| Provides an array containing only unique items from the input array                  | List<E>              |
+| `random()`  | Provides the random item of an array                                                 | Depends on the array |
+
+>>>>>>> Stashed changes
```
Comment (valfirst) [inline README.md]:
```suggestion
```

File: README.md [hunk 2]
```diff
@@ -73,6 +73,9 @@ The function output is dictated by the function itself.
 | `last()`    | Provides the last item of an array                                                   | Depends on the array |
 | `index(X)`  | Provides the item of an array of index: X, if the X is negative, take from backwards | Depends on the array |
 | `distinct()`| Provides an array containing only unique items from the input array                  | List<E>              |
+| `random()`  | Provides the random item of an array                                                 | Depends on the array |
```
Comment (valfirst) [inline README.md]:
```suggestion
| `random()`  | Provides one random item from the array                                                 | Depends on the array |
```

File: json-path/src/main/java/com/jayway/jsonpath/internal/function/sequence/Random.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+public class Random implements PathFunction {
+    @Override
+    public Object invoke(String currentPath, PathRef parent, Object model, EvaluationContext ctx, List<Parameter> parameters) {
+        if (ctx.configuration().jsonProvider().isArray(model)) {
+            Iterable<?> objects = ctx.configuration().jsonProvider().toIterable(model);
+            List<?> elements = StreamSupport.stream(objects.spliterator(), false)
+                                            .collect(Collectors.toList());
+
+            if (elements.isEmpty()) {
+                throw new JsonPathException("Aggregation function attempted random value on empty array");
```
Comment (valfirst) [inline json-path/src/main/java/com/jayway/jsonpath/internal/function/sequence/Random.java]:
```suggestion
                throw new JsonPathException("Aggregation function attempted to get random item from empty array");
```

File: json-path/src/main/java/com/jayway/jsonpath/internal/function/sequence/Random.java [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+            }
+
+            java.util.Random rand = new java.util.Random();
```
Comment (valfirst) [inline json-path/src/main/java/com/jayway/jsonpath/internal/function/sequence/Random.java]: `java.security.SecureRandom`

File: json-path/src/main/java/com/jayway/jsonpath/internal/function/sequence/Random.java [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+            return elements.get(rand.nextInt(elements.size()));
+        }
+        throw new JsonPathException("Aggregation function attempted random value using non array");
```
Comment (valfirst) [inline json-path/src/main/java/com/jayway/jsonpath/internal/function/sequence/Random.java]:
```suggestion
        throw new JsonPathException("Aggregation function attempted to get random item from not array");
```

**Review by valfirst — CHANGES_REQUESTED** [review]

File: json-path/src/main/java/com/jayway/jsonpath/internal/function/sequence/Random.java (hunk 4 above)
Comment (uarlouski) [reply json-path/src/main/java/com/jayway/jsonpath/internal/function/sequence/Random.java]: what is the reason to change it in this particular context? we do not care about security/predictability and just Random works much faster

**Review by valfirst — APPROVED** [review]

**Pull request merged**